src/xscen/diagnostics.py

```diff
@@ -7,6 +7,7 @@
 from copy import deepcopy
 from pathlib import Path
 from types import ModuleType
+from typing import Union
```

Comment:
Do you really need it? I think that it's mostly obsolete since Python 3.10.

Reply:
Got it, I also changed some xscen docstrings and replaced Union with the pipe |